reconcile/aws_version_sync/integration.py

```diff
@@ -228,8 +244,8 @@ def reconcile(
             current=external_resources_app_interface,
             desired=external_resources_aws,
             key=lambda r: r.key,
-            equal=lambda r1, r2: r1.resource_engine_version
-            == r2.resource_engine_version,
+            equal=lambda external_resources_app_interface, external_resources_aws: external_resources_app_interface.resource_engine_version
+            >= external_resources_aws.resource_engine_version,
```

Comment:
I didn't notice the `>=` check at the first glance, later realized it is a smart check here to ignore version bump in app interface. I think it's more explicit if do the check while iterating `diff_pair`:

```python
            if aws_resource.resource_engine_version > app_interface_resource.resource_engine_version:
```

so we make it clear that this integration only create mr to upgrade engine version.

Reply:
Yes, explicit is better than implicit. Applied in [42a62bd](https://github.com/app-sre/qontract-reconcile/pull/3940/commits/42a62bd6e0f86569e454dfc8f7b7775f8ae921b5)

Reply:
Do you want to revert back to equal https://github.com/app-sre/qontract-reconcile/pull/3940/files#diff-cbc1c6cbfb9a7d7a7394fa5e47bcfdcbf95f17e250cfe4aa4bfdffacd35cfd5aR247-R248?

Reply:
Applied in c1bb8bf